sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -54,16 +55,28 @@ def _recomendable_sale_order_lines_domain(self):
         """Domain to find recent SO lines."""
         start = datetime.now() - timedelta(days=self.months * 30)
         start = fields.Datetime.to_string(start)
-        other_sales = self.env["sale.order"].search(
-            [
-                (
-                    "partner_id",
-                    "child_of",
-                    self.order_id.partner_id.commercial_partner_id.id,
-                ),
-                ("date_order", ">=", start),
-            ]
-        )
+        if not self.use_delivery_address:
+            other_sales = self.env["sale.order"].search(
+                [
+                    (
+                        "partner_id",
+                        "child_of",
+                        self.order_id.partner_id.commercial_partner_id.id,
+                    ),
+                    ("date_order", ">=", start),
+                ]
+            )
+        else:
+            other_sales = self.env["sale.order"].search(
+                [
+                    (
+                        "partner_shipping_id",
+                        "child_of",
+                        self.order_id.partner_shipping_id.commercial_partner_id.id,
+                    ),
+                    ("date_order", ">=", start),
+                ]
+            )
```

Comment:
```suggestion
        partner_id = self.order_id.partner_id
        if self.use_delivery_address:
            partner_id = self.order_id.partner_shipping_id
        other_sales = self.env["sale.order"].search(
            [
                (
                    "partner_id",
                    "child_of",
                    partner_shipping_id.commercial_partner_id.id,
                ),
                ("date_order", ">=", start),
            ]
        )
```

Reply:
It isn't what I want to achieve... because with this solution you will allways search the sale orders by the partner_id... and I want to search it by partner_shipping_id when the option is selected

Reply:
Sorry, i have updated my previous suggestion.

Reply:
I think it's still incorrect... The domain is allways the same... When the option is selected it should search by partner_shipping_id and not by partner id... I don't know if I'm explaining myself well 😅

Reply:
Yes, of course, but these code:
```
partner_id = self.order_id.partner_id
if self.use_delivery_address:
    partner_id = self.order_id.partner_shipping_id
```
set "correct" `partner_id` to search after.

Reply:
I'd say it's still wrong @victoralmau

Reply:
@victoralmau I think that to respect the logic it would be:
```python
partner = self.order_id.partner_shipping_id if self.use_delivery_address else self.order_id.partner_id.commercial_partner_id
```
And then you can have the common filter

Reply:
Ok, thanks, i have update my initial suggestion about these change.